src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -681,13 +679,7 @@ private bool IsNounVerbMatch(CommandInfo command)
         /// </summary>
         private void AccumulateMatchingCommands()
         {
-            Collection<string> commandNames =
-                SessionStateUtilities.ConvertArrayToCollection<string>(this.Name);
-
-            if (commandNames.Count == 0)
-            {
-                commandNames.Add("*");
-            }
+            var commandNames = (Name != null && Name.Length != 0) ? Name : new string[] { "*" };
```

Comment:
Why not use `!string.IsNullOrEmpty()` here instead since that's what the method is for?

Reply:
```suggestion
            var commandNames = string.IsNullOrEmpty(Name) ? new string[] { "*" } : Name;
```